File: hw/gongora/U2/HW18Inheritance/java/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Amphibian.java


package ec.edu.espe.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Lucas Gongora
 */
public class Amphibian extends Animal {
    private String skinType;
    private String stage;
    
    public void hop(){
        System.out.println("The "+super.getName()+" is hopping");
    }
    
    public void transition(){
        System.out.println("The "+super.getName()+"is transforming");
    }

    @Override
    public String toString() {
        return "Amphibian{" + super.toString() + "skinType=" + skinType + ", stage=" + stage + '}';
    }

    public Amphibian(String skinType, String stage, int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.skinType = skinType;
        this.stage = stage;
    }

    
    
    
    

    /**
     * @return the skinType
     */
    public String getSkinType() {
        return skinType;
    }

    /**
     * @param skinType the skinType to set
     */
    public void setSkinType(String skinType) {
        this.skinType = skinType;
    }

    /**
     * @return the stage
     */
    public String getStage() {
        return stage;
    }

    /**
     * @param stage the stage to set
     */
    public void setStage(String stage) {
        this.stage = stage;
    }
    
    
}

File: hw/quiroga/U2/Hw18Exceptions/Inheritance_JS/javascript/ec/edu/espe/zoo/model/Amphibian.mjs
import { Animal } from "./Animal.mjs";


export class Amphibian extends Animal {
    #skinType;
    #stage;

    constructor(skinType, stage, id, name, bornOnDate, weight, sex, vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.#skinType = skinType;
        this.#stage = stage;
    }

    hop() {
        console.log(`The ${super.getName()} is hopping`);
    }

    transition() {
        console.log(`The ${super.getName()} is transforming`);
    }

    toString() {
        return `Amphibian{${super.toString()} skinType=${this.#skinType}, stage=${this.#stage}}`;
    }

    getSkinType() {
        return this.#skinType;
    }

    setSkinType(skinType) {
        this.#skinType = skinType;
    }

    getStage() {
        return this.#stage;
    }

    setStage(stage) {
        this.#stage = stage;
    }
}
